ideascube/configuration/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+from .errors import (
+    InvalidConfigurationValue,
+    InvalidStoredConfigurationValue,
+    NoSuchConfigurationKey,
+    NoSuchConfigurationNamespace,
+    )
+from .registry import REGISTRY
+
+
+class Configuration(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def _get_key_registry(namespace, key):
+        try:
+            namespace_registry = REGISTRY[namespace]
+
+        except KeyError:
+            raise NoSuchConfigurationNamespace(namespace)
+
+        try:
+            return namespace_registry[key]
+
+        except KeyError:
+            raise NoSuchConfigurationKey(namespace, key)
+
+    @classmethod
+    def get(cls, namespace, key):
+        key_registry = cls._get_key_registry(namespace, key)
+        expected_type = key_registry['python_type']
+
+        config = cls.objects.get(namespace=namespace, key=key)
+
+        if not isinstance(config.value, expected_type):
```

Comment:
I would agree with you on this point if it were possible to have an incorrect type stored in the DB. (i.e what we have currently in master)

But the `set` method enforces the same constraint.

The only way to have an invalid type stored into the db is if someone manually goes and modify the DB. And frankly, I don't think I want to defend against this.

With this PR we're providing a simple to use API that enforces type constraints, and with the next PR we'll provide a configuration tool that will be simple to use and enforce the same type constraints.

If an admin goes and modify the DB manually and break the server, which is harder to do than using the provided tools, then they seriously had it coming.

If someone looks at the sqlite db file like a binary blob and modifies random bits (changes 0's into 1's and the opposite) in it, then sqlite will just barf. That is because the correct thing to do is to use the sqlite API and it enforces certain constraints on how the 0's and 1's are layed out in the binary db file. Essentially, the sqlite contract is « use my API and I guarantee that you'll always get a correct DB you can interact with; if you mess with the DB file you're on your own, I don't want to have anything to do with that ».

I strongly feel that with the set of constraints we're introducing in this PR on the `Configuration` model, we should have a similar contract: « use our API and we guarantee that the types will always be correct; if you mess up with the configuration records in the DB you're on your own, we don't want to have anything to do with that ».

Now, we might need to introduce more constraints (and testing!) to be able to provide such a guarantee. But I **strongly** suggest that we try moving in the direction of such a contraints+guarantee contract, than trying to defend against things that just shouldn't happen in the first place.